include/onnxruntime/interface/framework/kernel.h

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <cassert>
+#include <numeric>
+#include "interface/common/data_types.h"
+#include "core/common/status.h"
+
+namespace onnxruntime {
+
+namespace interface {
+
+using TensorShape = std::vector<int64_t>;
```

Comment:
>TensorShape = std::vector<int64_t>; [](http://example.com/codeflow?start=6&length=35)

std vector might be expensive for TensorShape. could we use a interface for TensorShape, and hide the details like abseil / small vector implementation internally?

Reply:
That make sense.